app/lib/handle-input-change.js

```diff
@@ -0,0 +1,35 @@
+import DOMPurify from 'dompurify';
+
+export default function handleInputChange(e) {
+  /*
+  Using this module is a little odd.
+  Ensure that it's called in the context of a JSON-API Model instance.
+  */
+  if (typeof this.update !== 'function' || typeof this.emit !== 'function') {
+    throw new Error('Bind the handleInputChange function to a json-api-client Model instance')
+  }
+
+  let valueProperty = 'value';
+  switch (e.target.type) {
+    case 'checkbox': {
+      valueProperty = 'checked';
+      break;
+    }
+    case 'file': {
+      valueProperty = 'files';
+      break;
+    }
+  }
+
+  const value = e.target[valueProperty];
```

Comment:
if possibly redefined on line 27 should this be `let`?

Reply:
Good catch. I'm not sure what `dataset.jsonValue` is doing (other than marking the input value as JSON.) I couldn't find a reference to it anywhere else in the code.

Reply:
Looks like `target.dataset.jsonValue` is set by adding `data-json-value` to an input element, and it tells the change handler to parse the value as JSON.
https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/dataset

Reply:
I think it's used to parse strings to booleans and null values.
https://github.com/zooniverse/Panoptes-Front-End/search?q=data-json-value

Reply:
I've left it in the code, since `data-json-value` is still in use.